### ISSUE TITLE
Check if supplied GitHub account has read and write access to the repository

### ISSUE THREAD

**Issue by mdio** [czechboy0/Buildasaur#15]
We ran into a problem with our private repository and the GitHub user account used for Buildasaur had read only access to the repository.
Credential validation from Buildasaur was successful and Bots were created, integrations ran.
But posting the integration result to GitHub and adding the comment failed and it took us a while to figure out why.

It would be cool if Buildasaur could check for write access as well and at least warn that integration results cannot be published on GitHub without write access.

**Comment by czechboy0** [comment]
Good point, [right now](https://github.com/czechboy0/Buildasaur/blob/master/Buildasaur/NetworkUtils.swift#L25) Builda verifies access to the repository just by trying to read Pull Requests, which probably succeeded in your case. I guess I could try all the rights needed, by trying to create/delete a commit status and adding/deleting a comment on a PR. Or maybe there's a GitHub API call to verify rights of a token?

**Comment by mdio** [comment]
A few tests with the [repos GET endpoint](https://developer.github.com/v3/repos/#get) make me confident that the part with permissions is the key here. `push` (write) and `pull` (read) must be true in order to do everything Buildasaur does. Of course the requests need to be done with the token to reflect the permissions of the user which the token belongs to (but Buildasaur does that anyway).

``` js
"permissions": {
      "admin": false,
      "push": true,
      "pull": true
    }
```

**Comment by czechboy0** [comment]
@mdio :+1: Yeah that's it exactly. Feel free to send a pull request with the change if you want, it'd mean adding a method to [`GitHubServer`](https://github.com/czechboy0/Buildasaur/blob/master/BuildaGitServer/GitHubServer.swift) called `getRepository`, creating a new entity `Repository` and then just changing the verification call in [`checkAvailabilityOfGitHubWithCurrentSettingsOfProject`](https://github.com/czechboy0/Buildasaur/blob/master/Buildasaur/NetworkUtils.swift#L16) to use the permissions metadata to return whether Buildasaur has all the rights required. If not I'll tackle it in the next couple of days. 

Thanks for taking the time to look up the API call!

**Comment by mdio** [comment]
I'm afraid that I won't be able to implement that today and due to tomorrow being a public holiday I won't start before monday.
Also I'm just learning Swift but thanks to your detailed idea it looks doable.

**Comment by czechboy0** [comment]
No problem, I'll take a look at it over the weekend.

**Comment by czechboy0** [comment]
Fixed in 6642a0858d7e2521c4124c34201581d04927fc47. Let me know if this works for you, @mdio!
